internal/models/asset.go

```diff
@@ -0,0 +1,18 @@
+package models
+
+// AssetType is a custom type for asset types
+type AssetType string
+
+// Define constants for Asset Types
+const (
+	ChartType    AssetType = "Chart"
+	InsightType  AssetType = "Insight"
+	AudienceType AssetType = "Audience"
+)
+
+// Asset interface to be implemented by all asset types
+type Asset interface {
```

Comment:
Using interface as an asset provides a flexible way to handle various of asset types, which can be extended in the future adding new assets. Also with the interface we can define reusable and unified operations upon them (like get details etc.) and this way we reduce the complexity of having multiple structs with many different operations. 

A possible con that I can think of is that for simple applications with small number of assets and not many common and complex operations, this can add some overhead to the overall logic of the code.